prov/verbs/src/fi_verbs.c

```diff
@@ -938,29 +945,50 @@ static void fi_ibv_msg_ep_qp_init_attr(struct fi_ibv_msg_ep *ep,
 	}
 }
 
+static struct fi_info *fi_ibv_find_info(const char *fabric_name,
+		const char *domain_name)
+{
+	struct fi_info *info;
+
+	for (info = verbs_info; info; info = info->next) {
+		if ((domain_name && !strcmp(info->domain_attr->name, domain_name)) ||
+			(fabric_name && !strcmp(info->fabric_attr->name, fabric_name))) {
```

Comment:
last real change (no, really!) -- please change this check to something like:

if ((!domain || !strcmp(domain ...) && (!fabric || !strcmp(fabric ...))

I.e. if both the domain and fabric are provided, then both must match.  If neither are provider, we match on the item at the head of the list.